### ISSUE TITLE
Add a fix for CompileTimeErrorCode.NON_CONST_GENERATIVE_ENUM_CONSTRUCTOR

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/sdk#48479]
The diagnostic `CompileTimeErrorCode.NON_CONST_GENERATIVE_ENUM_CONSTRUCTOR ` should have a fix that would add the `const` keyword. The fix should probably only be applied if all of the fields are `final`.

**Comment by asashour** [comment]
It seems that [Enums can only declare final fields](https://dart.dev/tools/diagnostic-messages#non_final_field_in_enum).

Please look into https://dart-review.googlesource.com/c/sdk/+/239669

**Comment by bwilkerson** [comment]
> It seems that [Enums can only declare final fields](https://dart.dev/tools/diagnostic-messages#non_final_field_in_enum).

Yep. I didn't realize that when I opened the issue, but it makes sense given that enum constants are implicitly `const`.
